pandas/tests/io/parser/na_values.py

```diff
@@ -369,3 +369,27 @@ def test_no_na_filter_on_index(self):
         expected = DataFrame({"a": [1, 4], "c": [3, 6]},
                              index=Index([np.nan, 5.0], name="b"))
         tm.assert_frame_equal(out, expected)
+
+    def test_na_values_with_dtype_str_and_na_filter_true(self):
```

Comment:
can you parameterize this on na_filter (you will need to provide the nan_value as well in the parameterize as they are different)